src/api/spec/models/user_spec.rb

```diff
@@ -0,0 +1,95 @@
+require 'rails_helper'
+
+RSpec.describe User do
+  let(:admin_user) { create(:admin_user) }
+  let(:user) { create(:user) }
+  let(:input) { { 'Event::RequestCreate' => { source_maintainer: '1' } } }
+
+  it { is_expected.to validate_presence_of(:login).with_message('must be given') }
+  it { is_expected.to validate_length_of(:login).is_at_least(2).with_message('must have more than two characters.') }
+  it { is_expected.to validate_length_of(:login).is_at_most(100).with_message('must have less than 100 characters.') }
+
+  it { is_expected.to validate_presence_of(:email).with_message('must be given') }
+  it { is_expected.to allow_value('king@opensuse.org').for(:email) }
+  it { is_expected.to_not allow_values('king.opensuse.org', 'opensuse.org', 'opensuse').for(:email) }
+
+  it { is_expected.to validate_presence_of(:password).with_message('must be given') }
+  it { is_expected.to validate_length_of(:password).is_at_least(6).with_message('must have between 6 and 64 characters.') }
+  it { is_expected.to validate_length_of(:password).is_at_most(64).with_message('must have between 6 and 64 characters.') }
+
+  it 'sets password_hash_type to md5 by default' do
+    expect(user.password_hash_type).to eq('md5')
+  end
+
+  it 'sets state to unconfirmed by default' do
+    expect(user.state).to eq(User::STATES['unconfirmed'])
+  end
+
+  it 'validates uniqueness of login' do
+    user = create(:user)
+    invalid_user = build(:user, login: user.login)
```

Comment:
this is not only an "invalid" user... is more a "duplicated" username... maybe is better to change the variable name